main_test.go

```diff
@@ -261,35 +261,64 @@ func generateLiterals(ts *testscript.TestScript, neg bool, args []string) {
 
 	codePath := args[0]
 
-	// Add 100 randomly small literals.
 	var statements []ast.Stmt
+
+	// Add 100 randomly small literals.
 	for i := 0; i < 100; i++ {
-		literal := generateStringLit(1 + testRand.Intn(255))
-		statements = append(statements, &ast.AssignStmt{
-			Lhs: []ast.Expr{ast.NewIdent("_")},
-			Tok: token.ASSIGN,
-			Rhs: []ast.Expr{literal},
-		})
+		statements = append(statements, &ast.ExprStmt{
+			X: &ast.CallExpr{
+				Fun: &ast.SelectorExpr{
+					X:   &ast.Ident{Name: "fmt"},
+					Sel: &ast.Ident{Name: "Println"},
+				},
+				Args: []ast.Expr{generateStringLit(1 + testRand.Intn(255))},
+			},
+		},
+		)
 	}
-	// Add 5 huge literals, to make sure we don't try to obfuscate them.
+
+	// Add 5 huge literals, to make sure we obfuscate them fast.
 	// 5 * 128KiB is large enough that it would take a very, very long time
-	// to obfuscate those literals with our simple code.
+	// to obfuscate those literals with our too complex obfuscators.
 	for i := 0; i < 5; i++ {
-		literal := generateStringLit(128 << 10)
-		statements = append(statements, &ast.AssignStmt{
-			Lhs: []ast.Expr{ast.NewIdent("_")},
-			Tok: token.ASSIGN,
-			Rhs: []ast.Expr{literal},
-		})
+		statements = append(statements, &ast.ExprStmt{
+			X: &ast.CallExpr{
+				Fun: &ast.SelectorExpr{
+					X:   &ast.Ident{Name: "fmt"},
+					Sel: &ast.Ident{Name: "Println"},
```

Comment:
what is this for? I assume so that the compiler/linker don't drop the strings as unused? please document that.

I'd also use `println` rather than `fmt.Println`, then you don't need to worry about importing `fmt`.

Reply:
yep reason was that `go build` removed the literals. I'll add comments for the function.